examples/rvc-app/rvc-common/src/rvc-service-area-delegate.cpp

```diff
@@ -77,32 +108,89 @@ CHIP_ERROR RvcServiceAreaDelegate::Init()
 //*************************************************************************
 // command support
 
-bool RvcServiceAreaDelegate::IsSetSelectedAreasAllowed(MutableCharSpan statusText)
+bool RvcServiceAreaDelegate::IsSetSelectedAreasAllowed(MutableCharSpan & statusText)
 {
-    // TODO IMPLEMENT
-    return true;
+    return (mIsSetSelectedAreasAllowedDeviceInstance->*mIsSetSelectedAreasAllowedCallback)(statusText);
 };
 
 bool RvcServiceAreaDelegate::IsValidSelectAreasSet(const Commands::SelectAreas::DecodableType & req, SelectAreasStatus & areaStatus,
-                                                   MutableCharSpan statusText)
+                                                   MutableCharSpan & statusText)
 {
-    // TODO IMPLEMENT
+    // if req is empty list return true.
+    {
+        size_t reqSize;
+        if (req.newAreas.ComputeSize(&reqSize) != CHIP_NO_ERROR)
+        {
+            areaStatus = SelectAreasStatus::kInvalidSet; // todo Not sure this is the correct error to use here
+            CopyCharSpanToMutableCharSpan("error computing number of selected areas"_span, statusText);
```

Comment:
If `statusText` is too short for that string, this will leave `statusText` containing whatever happened to be on the stack up there, which will then get sent out to the client.  As in, this is a security bug that leaks stack memory on the wire.

Reply:
This is safe due to the fact that the size of all of these strings is smaller than `kMaxSizeStatusText`. I have added [this issue](https://github.com/project-chip/connectedhomeip/issues/35250) to updated the documentation wherever a `MutableCharSpan` is passed as a reference to say that if this is not edited successfully, it's size must be set to 0.